spyder/plugins/ipythonconsole/widgets/main_widget.py

```diff
@@ -1701,9 +1703,21 @@ def get_current_shellwidget(self) -> ShellWidget | None:
     @Slot(bool, str, str)
     @Slot(bool, bool)
     @Slot(bool, str, bool)
-    def create_new_client(self, give_focus=True, filename='', special=None,
-                          given_name=None, cache=True, initial_cwd=None,
-                          path_to_custom_interpreter=None):
+    def create_new_client(self, *args, **kwargs):
+        future = AsyncDispatcher(get_user_environment_variables)
+        future.connect(
+            AsyncDispatcher.QtSlot(
+                lambda future: self._create_new_client(
+                    future.result(), *args, **kwargs
```

Comment:
I think this should be the future, not the (blocking) result of the future...
```suggestion
                    future, *args, **kwargs
```